### PR TITLE
Fix WebSocket connection problem

### PR DESCRIPTION
**Pull request by miiu96**
- Reopen connection in case of an error with a `broken pipe` message or `connection reset by peer` message

### CODE REVIEW

**Review by iulianpascalau — DISMISSED** [review]

**Review by ssd04 — DISMISSED** [review]

**Review by ssd04 — DISMISSED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Review by ssd04 — DISMISSED** [review]

File: websocket/transceiver/transceiver_test.go [hunk 1]
```diff
@@ -117,15 +117,12 @@ func TestReceiver_ListenAndSendAck(t *testing.T) {
...
-	wg := &sync.WaitGroup{}
-	wg.Add(1)
-
 	count := 0
 	conn := &testscommon.WebsocketConnectionStub{
 		ReadMessageCalled: func() (int, []byte, error) {
 			time.Sleep(500 * time.Millisecond)
-			if count >= 1 {
-				wg.Done()
+			if count == 2 {
```
Comment (iulianpascalau) [inline websocket/transceiver/transceiver_test.go:124]: is this ok? Should you use a mutex to access the counter value?

Comment (miiu96) [reply websocket/transceiver/transceiver_test.go:124]: refactored

File: websocket/transceiver/transceiver_test.go [hunk 2]
```diff
@@ -117,15 +117,16 @@ func TestReceiver_ListenAndSendAck(t *testing.T) {
...
 	})
 
-	count := 0
+	count := uint64(0)
 	conn := &testscommon.WebsocketConnectionStub{
 		ReadMessageCalled: func() (int, []byte, error) {
 			time.Sleep(500 * time.Millisecond)
-			if count == 2 {
-				count++
+			atomic.LoadUint64(&count)
```
Comment (iulianpascalau) [inline websocket/transceiver/transceiver_test.go:124]: doesn't work like this

**Review by iulianpascalau — CHANGES_REQUESTED** [review]

File: websocket/transceiver/transceiver_test.go (hunk 2 above)
Comment (miiu96) [reply websocket/transceiver/transceiver_test.go]: changed

File: websocket/integrationTests/clientAndServer_test.go [hunk 3]
```diff
@@ -71,7 +69,6 @@ func TestStartServerAddClientAndCloseClientAndServerShouldReceiveClose(t *testin
 	_ = wsServer.SetPayloadHandler(&testscommon.PayloadHandlerStub{
 		ProcessPayloadCalled: func(payload []byte, topic string, version uint32) error {
 			require.Equal(t, []byte("test"), payload)
```
Comment (iulianpascalau) [inline websocket/integrationTests/clientAndServer_test.go:71]: there is a risk this won't be evaluated

Comment (miiu96) [reply websocket/integrationTests/clientAndServer_test.go:71]: fixed

**Review by iulianpascalau — APPROVED** [review]

**Review by ssd04 — APPROVED** [review]

**Pull request merged**